src/reducers/Nins.js

```diff
@@ -4,11 +4,23 @@ const ninState = {
   message: "",
   nin: "",
   rmNin: "",
-  nins: []
+  nins: [],
+  showNinAtProfile: false,
+  showNinAtIdentity: false
 };
 
 let ninsReducer = (state = ninState, action) => {
   switch (action.type) {
+    case actions.SHOW_NIN_PROFILE:
+      return {
+        ...state,
+        showNinAtProfile: !state.showNinAtProfile
+      };
+      case actions.SHOW_NIN_IDENTITY:
+        return {
+          ...state,
+          showNinAtIdentity: !state.showNinAtIdentity
+        };
```

Comment:
maybe I missed the obvious answer to this, but something I would ask myself if I was building this would be if it is necessary for the entire app to know that this data is showing or if it could just be a local state change in the relevant component.

Reply:
If we use local state, this will change to default (Hidden) as soon as you change section/Tab
The point with "Show/Hide" is to keep the state in the entire app.